### ISSUE TITLE
Operation Yaquina Bay

### ISSUE THREAD

**Issue by paulirish** [GoogleChrome/lighthouse#2146]
Named after the Yaquina Bay lighthouse which, _like a fast webpage that you can see it load quickly from start to finish_, is short enough that it doesn't take long to see the whole thing:
![image](https://cloud.githubusercontent.com/assets/39191/25690902/b8770a26-304a-11e7-9c4e-9a5c6fb20ec4.png)

----

OYB is about making Lighthouse itself fast. I mean.. a performance tool should probably be performant. 😉 

### Goals: 

1. Make the default 4 category run of lighthouse faster
2. Improve the run duration for users who want to run a single category

The full start-to-finish LH run of cnn.com takes 3 minutes. Oh my. Waiting for load of the first pass takes 25 seconds, so that's the absolute fastest we can make the entire LH run take right now.  

While cutting the time to 1/6th is a stretch **we should at least cut it in half**.

-------

While profiling LH runtimes, we uncovered a few issues:

1. The disabled cache for the entire run (#2089) slows down passes 3 and 4. 
1. The Styles gatherer is generally taking >10s on heavy publisher sites
1. Speedline is also going >10s on long pageloads 
1. Retrieving trace can take like 10s easily.
1. TTI(beta) / LoadFastEnough audits are costly (~5s) on larger traces.

I think we can do a few selective fixes here to drive our total run duration down pretty easily.

**Comment by patrickhulce** [comment]
Update on "TTI(beta) / LoadFastEnough audits are costly (~5s) on larger traces":

tl;dr remove old TTI (alpha) code before I/O will get us a few seconds, nothing can be done about TTFI in the short term since it's only ~20ms of computation after getting a trace model

For firstInteractive, getting the trace model accounts for roughly 99.5% of the time spent (3110ms spent in getting trace model, 17ms spent filtering to long tasks and computing the result) so if we do away with that and compute durations slices ourselves then we can likely get a pretty big win. However, we'd really just shift the cost to EIL since it also uses trace model. This seems an unlikely dependency to remove before I/O and likely to introduce bugs if we maintain ourselves, but I might investigate further if time permits.

For old TTI, getting the trace model/running speedline only accounts for ~66% of the time spent (4738ms to get trace model/speedline since trace model should be cached by then, 2422ms computing) so there's likely more room for improvement but killing that entirely should save us that time anyway

**Comment by patrickhulce** [comment]
> Retrieving trace can take like 10s easily.

FWIW, 5s of wait time is mandatory based on the definitions of the metrics. Most of my CNN runs have been about ~6-8s so it's not as big a candidate for improvement as we might like.

**Comment by brendankenny** [comment]
> we'd really just shift the cost to EIL since it also uses trace model

EIL only needs durations of the top-level tasks (plus start/end of first/last for clipping), so if we extract them ourselves it won't need a trace model either.

**Comment by patrickhulce** [comment]
> EIL only needs durations of the top-level tasks (plus start/end of first/last for clipping), so if we extract them ourselves it won't need a trace model either.

👍  Yeah just updated, thought it used something else off the trace model but realized the percentile stuff doesn't actually use anything but the durations

**Comment by paulirish** [comment]
Holy crap, we've made some huge progress here.

### Full run for cnn.com
 * 10 days ago: 169s
 * Today: 50s **(238% faster)**

### Full run for theverge.com
 * 10 days ago: 92s
 * Today: 53s **(73% faster)**

### Full run for paulirish.com
 * 10 days ago: 30
 * Today: 26s _(15% faster)_

### Full run for example.com
 * 10 days ago: 13s
 * Today: 12s _(8% 😛 )_

"Today's run" includes https://github.com/GoogleChrome/lighthouse/pull/2220 
Measured from the CLI:  `time lighthouse <url>`

nice work, everyone.  @patrickhulce, you made quick work of the slowness!


cc @pavelfeldman

**Comment by paulirish** [comment]
We'll call this a win.

🎉 🎉 🎉 🎉
